### ISSUE TITLE
Add MaxBufferedItemCount configuration option

### ISSUE THREAD

**Issue by tony-gutierrez** [Azure/azure-cosmos-js#255]
According to this page https://docs.microsoft.com/en-us/azure/cosmos-db/performance-tips, these are two fields used to tune performance, yet they do not seem to exist in this sdk?

**Comment by southpolesteve** [comment]
@tony-gutierrez `MaxBufferedItemCount` is not configurable in this SDK. I'll keep this open to track it.

`MaxConnectionLimit` only applies to .NET. In this SDK, you can pass a custom agent in the client constructor to get full control over the max number of connections and connection behavior. However, I did notice the defaults were set to 256 max sockets. I am not sure why, but I have removed it here: https://github.com/Azure/azure-cosmos-js/pull/278

If you would like to test it out now by passing a custom agent with the defaults except for `keepAlive`:

``` js
 const https = require('https')
 const agent = new https.Agent({ keepAlive: true })
 const client = new CosmosClient({ endpoint, agent })
```

**Comment by tony-gutierrez** [comment]
Cool, thanks.

**Comment by christopheranderson** [comment]
Resolved in #397 

Note, we didn't use maxBufferedItemCount in this PR, we just added a toggle that buffered pages per endpoint. Overall, this is more straightforward than understanding the correct number for maxBufferedItemCount. In theory, we could still add that as a separate feature, but would need to have a specific scenario where a certain number of items ought to be buffered.
